### PR TITLE
Delete jobs in pman using delete command

### PR DESCRIPTION
**Pull request by husky-parul**
Added env variables, volumes, and volumeMount to reaper module. This was missed in previous PR causing reaper to fail.

### CODE REVIEW

File: openshift/pman-openshift-template.json [hunk 1]
```diff
@@ -195,9 +195,27 @@
...
+                                    "name": "OPENSHIFTMGR_PROJECT",
+                                    "value": "radiology"
+                                  }],
+                                  "volumeMounts": [{
+                                    "name": "kubecfg-volume",
+                                    "mountPath": "/tmp/.kube/",
+                                    "readOnly": true
+                                  }]
                               }
-                          ],
```
Comment (danmcp) [inline openshift/pman-openshift-template.json]: Can you add a return here to be consistent on formatting.

File: openshift/pman-openshift-template.json [hunk 2]
```diff
@@ -195,9 +195,27 @@
...
+                                    "name": "OPENSHIFTMGR_PROJECT",
+                                    "value": "radiology"
+                                  }],
+                                  "volumeMounts": [{
+                                    "name": "kubecfg-volume",
+                                    "mountPath": "/tmp/.kube/",
+                                    "readOnly": true
+                                  }]
+                              }],
+                          "volumes": [{
```
Comment (danmcp) [inline openshift/pman-openshift-template.json]: formatting doesn't line up here

Comment (husky-parul) [reply openshift/pman-openshift-template.json]: @danmcp @ravisantoshgudimetla 
Using openshiftmgr.py to delete job.

File: pman/pman.py [hunk 3]
```diff
@@ -1350,11 +1350,62 @@ def t_status_process_container(self, *args, **kwargs):
...
+        
+    def t_delete_process_openshift(self,*args, **kwargs):
+        """
+        Delete job and related resources (pods & pvc) from openshift
+        """
+        jid = ''
+        for k,v in kwargs.items():
+            if k == 'request' and v['action'] == 'delete' :     jid = v['meta']['value']
+        d_json  = self.get_openshift_manager().state(jid)
+        status = ''
```
Comment (danmcp) [inline pman/pman.py]: = None

**Comment by danmcp** [comment]
@husky-parul There are a couple of nit comments.  Also is the PR ready/tested?  It still says DO NOT MERGE in the title.

**Comment by danmcp** [comment]
@husky-parul There are still a couple of unresolved comments on this pull.  Let me know when you are ready for another review.

**Comment by rudolphpienaar** [comment]
Good to merge? @danmcp ?

**Comment by danmcp** [comment]
@rudolphpienaar There are still a couple of unresolved comments on this one.

**Comment by husky-parul** [comment]
@danmcp @ravisantoshgudimetla 
All changes add. Ready for review.

**Comment by danmcp** [comment]
Seems relevant: https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#clean-up-finished-jobs-automatically

**Comment by husky-parul** [comment]
> Seems relevant: https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#clean-up-finished-jobs-automatically
This requires adding `ttlSecondsAfterFinished` in job spec in  `openshiftmgr.py`

**Comment by danmcp** [comment]
@husky-parul And does it theoretically solve the problem or would something still be missing?

**Comment by husky-parul** [comment]
> @husky-parul And does it theoretically solve the problem or would something still be missing?
If this works, we can do away with cron jobs to delete old jobs (completed/failed).  I am testing it. 
Should we remove `delete` command from pman as all the jobs will be deleted eventually when their TTL expires?

**Comment by danmcp** [comment]
@husky-parul We would still need delete for tests unless the ttl is really short

**Comment by ravisantoshgudimetla** [comment]
I think this is enabled in kube 1.12(that alpha), so might not work with OpenShift 3.11 yet.(4.0 is based on 1.12 but currently rebase to kube 1.12 is in progress and we cannot see this feature in OpenShift till it graduates to beta in kube upstream.)

**Comment by husky-parul** [comment]
> I think this is enabled in kube 1.12(that alpha), so might not work with OpenShift 3.11 yet.(4.0 is based on 1.12 but currently rebase to kube 1.12 is in progress and we cannot see this feature in OpenShift till it graduates to beta in kube upstream.)

@danmcp @ravisantoshgudimetla setting TTL is not deleting jobs as of now. Should we keep cron jobs ?

**Comment by husky-parul** [comment]
Added `ttlSecondsAfterFinished` for Openshift 4.0 support.

**Comment by danmcp** [comment]
@husky-parul Everything tested and ready to merge?

File: openshift/pman-reaper/reaper.py [hunk 4]
```diff
@@ -62,9 +62,12 @@ def delete_jobs(self):
         jobs : <List> list of jobs to be deleted
         """
         jobs = self.get_jobs(self.age)
+        print('Jobs queued for delete: ',jobs)
         for job in jobs:
             try: 
-                self.kube_v1_batch_client.delete_namespaced_job(job, body={}, namespace=self.project, propagation_policy='Forward')
+                body = k_client.V1DeleteOptions(propagation_policy='Background')
+                self.kube_v1_batch_client.delete_namespaced_job(job, body=body, namespace=self.project)
+                print('Deleted job: ',job)
```
Comment (danmcp) [inline openshift/pman-reaper/reaper.py]: Nit: add space after ,

File: pman/pman.py [hunk 5]
```diff
@@ -1350,11 +1350,61 @@ def t_status_process_container(self, *args, **kwargs):
...
+        If yes, call t_delete_process_openshift to delete the job.
+        Pending implementation for other container environment.
+        """
+        status = jid = ''
+        if self.container_env == 'openshift':
+            self.dp.qprint('Processing openshift....')
+            try:
+                d_containerStatus       =   self.t_delete_process_openshift(*args, **kwargs)
+                status                  =   d_containerStatus['status']
+                jid                  =   d_containerStatus['jid']
```
Comment (danmcp) [inline pman/pman.py]: Nit: Doesn't aligh

File: pman/pman.py [hunk 6]
```diff
@@ -1350,11 +1350,61 @@ def t_status_process_container(self, *args, **kwargs):
             'logs':             str_logs,
             'currentState':     d_ret['d_process']['currentState']
         }
+    
+    def t_delete_process(self,*args, **kwargs):
+        """
+        Deletes existing jobs. Checks if container enviornment is openshift.
```
Comment (danmcp) [inline pman/pman.py]: Spelling: environment

Comment (danmcp) [reply pman/pman.py]: Also case on OpenShift

File: pman/pman.py [hunk 7]
```diff
@@ -1350,11 +1350,61 @@ def t_status_process_container(self, *args, **kwargs):
...
+                "status":   status
+            }
+        else:
+            # TODO: Implement delete job for other container environment here.
+            pass
+
+        
+    def t_delete_process_openshift(self,*args, **kwargs):
+        """
+        Delete job and related resources (pods & pvc) from openshift
```
Comment (danmcp) [inline pman/pman.py]: Case on OpenShift

File: pman/pman.py [hunk 8]
```diff
@@ -1350,11 +1350,61 @@ def t_status_process_container(self, *args, **kwargs):
...
+                'job_id' : jid,
+                'status' : status
+            }
+            return {
+                "d_ret":    d_ret,
+                "status":   status
+            }
+        else:
+            # TODO: Implement delete job for other container environment here.
+            pass
```
Comment (danmcp) [inline pman/pman.py]: What's this?

File: pman/pman.py [hunk 9]
```diff
@@ -1350,11 +1350,61 @@ def t_status_process_container(self, *args, **kwargs):
...
+
+        
+    def t_delete_process_openshift(self,*args, **kwargs):
+        """
+        Delete job and related resources (pods & pvc) from Openshift
```
Comment (danmcp) [inline pman/pman.py]: OpenShift

File: pman/pman.py (hunk 8 above)
Comment (danmcp) [reply pman/pman.py]: @husky-parul Aren't we removing that case as soon as the openshift path is done?

**Comment by danmcp** [comment]
LGTM  @rudolphpienaar Please review and merge.

**Pull request merged**